psl-core/src/main/java/org/linqs/psl/model/rule/Rule.java

```diff
@@ -38,4 +38,14 @@
 	public int groundAll(AtomManager atomManager, GroundRuleStore groundRuleStore);
 
 	public boolean isWeighted();
+
+	/**
+	 * @return a name for the rule
+	 */
+	public String getName();
```

Comment:
I don't like how this can return null.
Perhaps we should think about defaulting a name to nameless rules.